### ISSUE TITLE
Suggestions `nameaddon`: Lists and Crossrefs

### ISSUE THREAD

**Issue by sievertst** [plk/biblatex#1238]
First of all I would like to thank you for your work and dedication. Though I have only recently switched to using your package, it has been an absolute joy to work with and the level of customisation is simply astonishing.

Nevertheless, I do have one minor suggestion concerning the way the `nameaddon` field is treated. I work in East Asian studies where we have to give authors’/editors’ names in the original spelling all the time since it can’t be inferred from the romanisation. This is a bit problematic in BibLaTeX if you try to put it directly in the `author`/`editor` fields because you have to switch the position of the native spelling around depending on where the citation style "first last" and "last, first" orders if you want to keep the native spelling last (using the "Jr." position kind of works but is still suboptimal). With your style the `nameaddon` works like a charm – at least as long as there is only one `author`/`editor`. If there is more than one, the current behaviour is that the content of `nameaddon` is simply put at the end of the list. For example assuming the following input:

```bibtex
author={Miyamoto, Shigeru and Kojima, Hideo},
nameaddon={宮本茂 and 小島秀夫},
```

Currently this gives (assuming no format changes): Miyamoto, Shigeru and Hideo Kojima [宮本茂 and 小島秀夫]. What I would hope for instead would be treating `and` as a list separator and zipping the two lists: Miyamoto, Shigeru [宮本茂] and Hideo Kojima [小島秀夫].

Of course, this approach runs into a slight issue if only some of the authors/editors have a character name, but not all. In such a case I suggest doing the following:

```bibtex
author={Spielberg, Steven and Kurosawa, Akira},
nameaddon={{} and 黒沢明},
```

Intended output: Spielberg, Steven and Akira Kurosawa [黒沢明]

As a second suggestion, it would also be very useful if there was an option to add the `nameaddon` on the `author`/`editor` of a `crossref`ed work in the bibliography entry for the `crossref`ing work. That would provide a neat way of getting around the restriction that there is only one `nameaddon` field but potentially more than one field containing names. I assume this is probably already doable if I do a little `\renewcommand`ing but you can always dream of having it more convenient, right? ;)

Again, thanks a lot for your awesome package, keep up the good work!

**Comment by sievertst** [comment]
Sorry, I seem to have followed the wrong link and thought I had ended up at the style’s repo rather than the overall BibLaTeX one. Closing because my issue is style-specific.

Still, my thanks of course also goes to the contributors of BibLaTeX as well. You’re awesome!

**Comment by moewew** [comment]
The standard `biblatex` data model also has a `nameaddon` field, so let me quickly make a general point.

Even though the `biblatex` documentation suggests `nameaddon` might be used to add a pen name or pseudonym to a name, it is actually ill suited to this task, because `nameaddon` is just printed after the full author field. It cannot naturally deal with list situations where there are several names. Technically `nameaddon` isn't even a list field itself, but even if it were, "interleaved" printing of list fields is a pain.

For most intents an purposes where you'd want to add more stuff to a particular name that only applies to that name (a pseudonym, CJK version of the name), the extended name format or field annotations seem to be much more promising. 

https://tex.stackexchange.com/q/233053/35864 might be interesting for your specific use case. (https://tex.stackexchange.com/q/479460/35864 uses a similar approach for pseudonyms.)

**Comment by sievertst** [comment]
Thank you very much for your help! Your suggestion in https://tex.stackexchange.com/q/233053/35864 got me 99% where I wanted to be and gave me the tools to adjust the parts that I did need changed. Many thanks!

It is a little unfortunate that the documentation suggests a use that is not actually ideal. Maybe it could prevent long term confusion if it were changed to indicate this, at least in my eyes not entirely intuitive, limitation.
